include/net/nrf_cloud_location.h

```diff
@@ -0,0 +1,148 @@
+ */
+typedef void (*nrf_cloud_location_response_t)(const struct nrf_cloud_location_result *const pos);
+
+/**@brief Perform an nRF Cloud location request over MQTT.
+ *
+ * @param cells_inf Pointer to cell info. The required network parameters are
+ *                  cell identifier, mcc, mnc and tac. The required neighboring
+ *                  cell parameters are E-ARFCN and physical cell identity.
+ *                  The parameters for time diff and measurement time are not used.
+ *                  The remaining parameters are optional; including them may improve
+ *                  location accuracy. To omit a request parameter, use the appropriate
+ *                  define in @ref nrf_cloud_location_cell_omit.
+ *                  Can be NULL if Wi-Fi info is provided.
+ * @param wifi_inf Pointer to Wi-Fi info. The MAC address is the only required parameter
+ *                 for each item.
+ *                 To omit a request parameter, use the appropriate
+ *                  define in @ref nrf_cloud_location_wifi_omit.
+ *                 Can be NULL if cell info is provided.
+ * @param request_loc If true, cloud will send location to the device.
+ *                    If false, cloud will not send location to the device.
+ * @param cb Callback function to receive parsed location result. Only used when
+ *           request_loc is true. If NULL, JSON result will be sent to the cloud event
+ *           handler as an NRF_CLOUD_EVT_RX_DATA_LOCATION event.
+ * @retval 0       Request sent successfully.
+ * @retval -EACCES Cloud connection is not established; wait for @ref NRF_CLOUD_EVT_READY.
+ * @retval -EDOM The number of access points in the Wi-Fi-only request was smaller than
+ *               the minimum required value NRF_CLOUD_LOCATION_WIFI_AP_CNT_MIN.
+ * @return A negative value indicates an error.
+ */
+int nrf_cloud_location_request(const struct lte_lc_cells_info *const cells_inf,
+			       const struct wifi_scan_info *const wifi_inf,
+			       const bool request_loc, nrf_cloud_location_response_t cb);
+#endif /* CONFIG_NRF_CLOUD_MQTT */
+
+/**@brief Get the reference to a cJSON object containing a location request.
+ *
+ * @param cells_inf Pointer to cell info; can be NULL if Wi-Fi info is provided.
+ * @param wifi_inf Pointer to Wi-Fi info; can be NULL if cell info is provided.
+ * @param request_loc If true, cloud will send location to the device.
+ *                    If false, cloud will not send location to the device.
+ * @param req_obj_out Pointer used to get the reference to the generated cJSON object.
+ *
+ * @retval 0 If successful.
+ * @retval -EDOM The number of access points in the Wi-Fi-only request was smaller than
+ *               the minimum required value NRF_CLOUD_LOCATION_WIFI_AP_CNT_MIN.
+ * @return A negative value indicates an error.
+ */
+int nrf_cloud_location_request_json_get(const struct lte_lc_cells_info *const cells_inf,
+					const struct wifi_scan_info *const wifi_inf,
```

Comment:
We don't get a link in the documentation for `struct wifi_scan_info` as that structure is in a defgroup which is likely not anywhere in the documentation.

Reply:
The linking to `wifi_scan_info` doesn't exist in the API documentation. And I'm not sure if this is anywhere in the documentation.

Reply:
@peknis how can we get this new file and struct into the documentation?
i added the file `include/net/wifi_location_common.h` to the .rst but that did not do it.

Reply:
I guess you'd need to have `wifi_location_common` defgroup included somewhere like you have `nrf_cloud_location`:
```
.. doxygengroup:: nrf_cloud_location
   :project: nrf
   :members:
```
However, I don't know the exact syntax to add second defgroup and don't have time to figure that out.

Reply:
that took care of it, thanks @trantanen